pkg/util/kubernetes/autoscalers/processor.go

```diff
@@ -131,14 +137,54 @@ func (p *Processor) ProcessWPAs(wpa *v1alpha1.WatermarkPodAutoscaler) map[string
 	return externalMetrics
 }
 
+func makeChunks(batch []string) (chunks [][]string) {
+	for i := 0; i < len(batch); i += chunkSize {
+		if i+chunkSize > len(batch) {
+			chunks = append(chunks, batch[i:])
+			break
+		}
+		chunks = append(chunks, batch[i:i+chunkSize])
+	}
+	return chunks
+}
+
 // validateExternalMetric queries Datadog to validate the availability and value of one or more external metrics
 func (p *Processor) validateExternalMetric(emList map[string]custommetrics.ExternalMetricValue) (processed map[string]Point, err error) {
-	var batch []string
+	batch := []string{}
 	for _, e := range emList {
 		q := getKey(e.MetricName, e.Labels)
 		batch = append(batch, q)
 	}
-	return p.queryDatadogExternal(batch)
+	chunks := makeChunks(batch)
+	log.Tracef("List of batches %v", chunks)
+
+	// we have a number of chunks with `chunkSize` metrics.
+	responses := make(chan map[string]Point, len(batch))
+	var errors []error
+	processed = make(map[string]Point)
+	var waitResp sync.WaitGroup
+	for _, c := range chunks {
+		waitResp.Add(1)
+		go func(chunk []string) {
+			defer waitResp.Done()
+			resp, err := p.queryDatadogExternal(chunk)
+			if err != nil {
+				errors = append(errors, err)
```

Comment:
you are appending to an errors slice from multiple go routines, does test with race flag this as a data race?

Reply:
Yes, I think it can be an issue. 
A possible solution could be adding the possible error also the channel
```golang
type queryResponse struct {
     metrics map[string]Point
     err error
}
responses := make(chan queryResponse, len(batch))
```

Reply:
good point, I only have one test case for errors, so I missed this.